cmd/monaco/download/download_configs.go

```diff
@@ -179,42 +179,60 @@ func downloadConfigs(apis api.APIs, opts downloadConfigsOptions) (project.Config
 
 	c = client.LimitClientParallelRequests(c, opts.concurrentDownloadLimit)
 
-	apisToDownload := getApisToDownload(apis, opts.specificAPIs)
-	if len(apisToDownload) == 0 {
-		return nil, fmt.Errorf("no APIs to download")
-	}
-
 	configObjects := make(project.ConfigsPerType)
 
-	// download specific APIs only
-	if len(opts.specificAPIs) > 0 {
-		log.Debug("APIs to download: \n - %v", strings.Join(maps.Keys(apisToDownload), "\n - "))
-		c := classic.DownloadAllConfigs(apisToDownload, c, opts.projectName)
-		maps.Copy(configObjects, c)
+	if shouldDownloadAPIs(opts) {
+		classicCfgs, err := downloadClassicConfigs(c, apis, opts.specificAPIs, opts.projectName)
+		if err != nil {
+			return nil, err
+		}
+		maps.Copy(configObjects, classicCfgs)
 	}
 
-	// download specific settings only
-	if len(opts.specificSchemas) > 0 {
-		log.Debug("Settings to download: \n - %v", strings.Join(opts.specificSchemas, "\n - "))
-		s := settings.Download(c, opts.specificSchemas, opts.projectName)
-		maps.Copy(configObjects, s)
+	if shouldDownloadSettings(opts) {
+		settingsObjects := downloadSettings(c, opts.specificSchemas, opts.projectName)
+		maps.Copy(configObjects, settingsObjects)
 	}
 
-	// return specific download objects
-	if len(opts.specificSchemas) > 0 || len(opts.specificAPIs) > 0 {
-		return configObjects, nil
+	return configObjects, nil
+}
+
+// shouldDownloadAPIs returns true unless onlySettings or specificSchemas but no specificAPIs are defined
+func shouldDownloadAPIs(opts downloadConfigsOptions) bool {
```

Comment:
let's rename this to shouldDownloadClassicConfigs

Reply:
rebased and amended to rename this